[WIP] Add rotation moves + make a new folder for path changes

asinghvi17 opened the pull request:
This PR moves each path change into its own file for simplicity / searchability.  It will also add three new moves:
- `RotateCamera(duration, axis, angle)` - rotate the camera around `axis`
- `OrbitCamera(duration, axis, angle)` - similar to `RotateCamera` in implementation, but rotates the camera around its `lookat` while keeping that `lookat` constant.
- `CompoundMove`: apply multiple moves simultaneously, so you can e.g. rotate the camera while translating.  Moves are applied in order.

asinghvi17 committed:
Refactor all current path changes to their own folder

asinghvi17 committed:
Add Rotations.jl

lazarusA commented:
I started using this package. Is really convenient, great. However, the functions defined in this PR are definitely something I'm kinda doing in a more cumbersome way 😄 .  How can I help to make this go into master ?

asinghvi17 commented:
More use cases would be nice, I wrote these path changes for myself but not 100% sure how people are using them / if there is a more general strategy!

asinghvi17 committed:
Merge branch 'main' into as/moves

asinghvi17 committed:
Add a slerp (spherical interpolation) move for orbits etc.

This is still pretty hacky, but it works.  The idea is that you can slerp every aspect of the view state, and you can also center it about some point so that you can rotate about that point.

This is not meant to be necessarily immediately user-usable, but rather a more foundational move for e.g. orbiting around a point or rotating a camera.  You can use e.g. Rotations.AngleAxis to set the final point and then this will go there but on the surface of a sphere parameterized by the original and final points, and the center point.

asinghvi17 committed:
remove show
